CMakeLists.txt

```diff
@@ -437,10 +439,16 @@ while(library_variants_copy)
         # system doesn't recognize every possible Arm architecture
         # version. So mostly we just say 'arm' and control the arch
         # version via -march=armv7m (or whatever).
-        if(target_arch STREQUAL "armv6m")
-            # An exception is armv6m, which compiler-rt expects to see
-            # in the triple because that's where it looks to decide
-            # whether to use the v6-M specific assembly sources.
+        # Exceptions are architectures pre-armv7, which compiler-rt expects to
+        # see in the triple because that's where it looks to decide whether to
+        # use specific assembly sources.
+        if(target_arch STREQUAL "armv4t")
+            set(target "armv4t-none-eabi")
+        elseif(target_arch STREQUAL "armv5te")
+            set(target "armv5te-none-eabi")
+        elseif(target_arch STREQUAL "armv6")
+            set(target "armv6-none-eabi")
+        elseif(target_arch STREQUAL "armv6m")
```

Comment:
```suggestion
        if(target_arch MATCHES "^armv[4-6]")
            set(target "${target_arch}-none-eabi")
```

Reply:
Right, unfortunately I'm getting a linker error on all platforms when compiling the picolibc tests (on otherwise clean repo) which I still need to get to the bottom of, so haven't been able to test this yet. Turns out machine <> cpu compatibility checking in QEMU didn't work like I expected. Looking closer, there's a Canon board that works with v5te (as well as the generic board I expect), but I'll test it first before patching the patch.

Reply:
fixed v5te picolibc tests. was an issue with setting up the men regions for appropriate v5te board.